test/test_unary_ufuncs.py

```diff
@@ -1,12 +1,151 @@
+        imag = torch.rand(size, device=device, dtype=float_dtype) * 18 - 9
+        imag = _maybe_clamp(imag, -9, low, 10, high)
+        return torch.complex(real, imag)
+
+
+# Returns an iterable of contiguous tensors with the same storage on the requested
+#   device and with the requested dtype. The iterable will include an empty
+#   tensor, scalar (0-dim) tensors, small 1D tensors, a medium 1D tensor, and
+#   a larger 2D tensor. The tensors will "cover" the values in vals, and if
+#   vals is None then dtype-specific values will be substituted.
+#   The tensors' elements, except for those specified in vals, will be in the
+#   domain [low, high), if specified.
+#   If include_large_values is False then larger values, like 500+, then the
+#     default values will not include larger values, like 500+.
+#   If include_extremal_values is False, then the default values will not include
+#     NaN, positive infinity, or negative infinity.
+#
+def generate_numeric_tensors(device, dtype, *,
+                             domain=(None, None),
+                             include_large_values=True,
+                             include_extremal_values=True):
+    medium_length = 812
+    large_size = (1029, 917)
+    offset = 63
+
+    assert large_size[1] > (medium_length + offset)
+    assert medium_length % 4 == 0
+
+    # Special-cases bool
+    if dtype is torch.bool:
+        tensors = (torch.empty(0, device=device, dtype=torch.bool),
+                   torch.tensor(True, device=device),
+                   torch.tensor(False, device=device),
+                   torch.tensor((True, False), device=device),
+                   _make_tensor((medium_length,), device=device, dtype=dtype, low=None, high=None),
+                   _make_tensor(large_size, device=device, dtype=dtype, low=None, high=None))
+        return tensors
+
+    # Acquires dtype-specific vals
+    if dtype.is_floating_point or dtype.is_complex:
+        large_vals = _large_float_vals if include_large_values else tuple()
+        extremals = _float_extremals if include_extremal_values else tuple()
+        vals = _float_vals + large_vals + extremals
+
+        # Converts float -> complex vals if dtype is complex
+        if dtype.is_complex:
+            vals = tuple(complex(x, y) for x, y in product(vals, vals))
+    elif dtype is torch.uint8:
+        vals = _unsigned_int_vals
+    else:  # dtypes is a signed integer type
+        assert dtype in (torch.int8, torch.int16, torch.int32, torch.int64)
+        large_vals = _large_int_vals if include_large_values else tuple()
+        vals = _int_vals + large_vals
+
+    assert len(vals) < medium_length
 
+    # Constructs the large tensor containing vals
+    large_tensor = _make_tensor(large_size, device=device, dtype=dtype, low=domain[0], high=domain[1])
+
+    # Inserts the vals at an odd place
+    large_tensor[57][offset:offset + len(vals)] = torch.tensor(vals, device=device, dtype=dtype)
```

Comment:
This is a little dangerous, because the rest of tensor has been initialized randomly and the contents of the other elements within a vector instruction can affect the numerics of the values you are interested in. A good recounting of a bug that afflicted us a while back is https://github.com/pytorch/pytorch/issues/32502

Reply:
Isn't that good to test for, though?

Reply:
Well, I mean, I generally agree we should test handling at weird offsets and sizes, to make sure we exercise all of the relevant vectorization codepaths. But if you are specifically trying to catch bugs like #32502 you'd probably write tests differently than this. 

Whether or not the usage pattern here is dangerous depends, I guess, on how exactly these values are used. But if you used these inputs to do something like "doing an operation elementwise is bit-for-bit the same as doing it vectorized" it would be quite wrong to test this case (since it's not actually true that these are equivalent :)

Reply:
I agree with your thinking and maybe we will want to tweak this in the future to test finer properties. But given the current tests I think we're OK.